plugin/storage/es/samplingstore/storage.go

```diff
@@ -0,0 +1,199 @@
+	}
+	return dbmodel.ToThroughputs(retSamples), nil
+}
+
+func (s *SamplingStore) InsertProbabilitiesAndQPS(hostname string,
+	probabilities model.ServiceOperationProbabilities,
+	qps model.ServiceOperationQPS,
+) error {
+	ts := time.Now()
+	writeIndexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, ts)
+	val := dbmodel.ProbabilitiesAndQPS{
+		Hostname:      hostname,
+		Probabilities: probabilities,
+		QPS:           qps,
+	}
+	s.writeProbabilitiesAndQPS(writeIndexName, ts, val)
+	return nil
+}
+
+func (s *SamplingStore) GetLatestProbabilities() (model.ServiceOperationProbabilities, error) {
+	ctx := context.Background()
+	indices, err := s.getLatestIndices(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest indices: %w", err)
+	}
+	searchResult, err := s.client().Search(indices...).
+		Size(s.maxDocCount).
+		IgnoreUnavailable(true).
+		Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for Latest Probabilities: %w", err)
+	}
+	lengthOfSearchResult := len(searchResult.Hits.Hits)
+	if lengthOfSearchResult == 0 {
+		return nil, nil
+	}
+
+	var latestProbabilities dbmodel.TimeProbabilitiesAndQPS
+	latestTime := time.Time{}
+	for _, hit := range searchResult.Hits.Hits {
+		var data dbmodel.TimeProbabilitiesAndQPS
+		if err = json.Unmarshal(*hit.Source, &data); err != nil {
+			return nil, fmt.Errorf("unmarshalling documents failed: %w", err)
+		}
+		if data.Timestamp.After(latestTime) {
+			latestTime = data.Timestamp
+			latestProbabilities = data
+		}
+	}
+	return latestProbabilities.ProbabilitiesAndQPS.Probabilities, nil
+}
+
+func (s *SamplingStore) writeThroughput(indexName string, ts time.Time, throughputs []*model.Throughput) {
+	s.client().Index().Index(indexName).Type(throughputType).
+		BodyJson(&dbmodel.TimeThroughput{
+			Timestamp:  ts,
+			Throughput: dbmodel.FromThroughputs(throughputs),
+		}).Add()
+}
+
+func (s *SamplingStore) writeProbabilitiesAndQPS(indexName string, ts time.Time, pandqps dbmodel.ProbabilitiesAndQPS) {
+	s.client().Index().Index(indexName).Type(probabilitiesType).
+		BodyJson(&dbmodel.TimeProbabilitiesAndQPS{
+			Timestamp:           ts,
+			ProbabilitiesAndQPS: pandqps,
+		}).Add()
+}
+
+func (s *SamplingStore) getLatestIndices(ctx context.Context) ([]string, error) {
+	now := time.Now().UTC()
+	for {
+		indexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, now)
+		exists, err := s.client().IndexExists(indexName).Do(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if exists {
+			return []string{indexName}, nil
+		}
+		now = now.Add(-time.Hour * 24)
+	}
+}
+
+func (s *SamplingStore) getReadIndices(start, end time.Time) []string {
+	var indices []string
+	firstIndex := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, start)
+	currentIndex := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, end)
+	for currentIndex != firstIndex {
+		indices = append(indices, currentIndex)
+		end = end.Add(-24 * time.Hour)
```

Comment:
similar to above, why 24hrs? How does this logic look in our main ES Span Store?

Reply:
in es span store, it is decided by this fn.
```go
func (c *Configuration) GetIndexRolloverFrequencySpansDuration() time.Duration {
	if c.IndexRolloverFrequencySpans == "hour" {
		return -1 * time.Hour
	}
	return -24 * time.Hour
}
```

Reply:
implemented the same here